sqlglot/dialects/snowflake.py

```diff
@@ -520,6 +520,8 @@ class Parser(parser.Parser):
         }
 
         SHOW_PARSERS = {
+            "DATABASES": _show_parser("DATABASES"),
+            "TERSE DATABASES": _show_parser("DATABASES"),
```

Comment:
I think you need to update `_parse_show_snowflake`, because this doesn't handle the following clause:

```
[ WITH PRIVILEGES <object_privilege> [ , <object_privilege> [ , ... ] ] ]
```

Today this is included in the `Command` string, so there's no parsing issue. Let's make sure this is tested as well.